### PR TITLE
[Checklist] Dynamic Type section.

### PR DESCRIPTION
**Pull request by willlarche**
#### Don't forget:
- [ ] Identify the component the PR relates to in brackets in the title. ```[Buttons] Updated documentation```
- [ ] Link to GitHub issues it solves. ```closes #1234```
- [ ] Sign the CLA bot. You can do this once the pull request is submitted.

[Contributing](./contributing/README.md#pull-requests) has more information and tips for a great
pull request.

### CODE REVIEW

File: contributing/checklist.md [hunk 1]
```diff
@@ -174,6 +174,21 @@ See Apple's [Accessibility Programming Guide for iOS](https://developer.apple.co
 1. Test your control on a device in VoiceOver mode and ensure the bahavior is at least as robust as UIKit.
 1. Enter YES, NO or N/A
 
+### Dynamic Type Support (If possible)
+
+
+Any component that has text, interacts with text, or lays itself out in relation to text must implement Dynamic Type.
```
Comment (ianegordon) [inline contributing/checklist.md]: s/must implement/should implement/
(We may decide some components should not for design / feasibility reasons)

File: contributing/checklist.md [hunk 2]
```diff
@@ -174,6 +174,21 @@ See Apple's [Accessibility Programming Guide for iOS](https://developer.apple.co
...
+See Apple's [Building Apps with Dynamic Type video](https://developer.apple.com/videos/play/wwdc2017/245/) for further information.
+
+1. Set all fonts with a text style enum like `MDCFontTextStyle` or `UIFontTextStyle`, or via `UIFontMetrics`.
+1. Implement `mdc_adjustsFontForContentSizeCategory` as a `BOOL` property.
+1. Add code that listens to and reacts to the `UIContentSizeCategoryDidChangeNotification`.
+1. Never layout code around text with 'magic numbers'. Instead, use a dynamic layout like Apple's auto layout.
+1. Make sure labels have an appropriate overflow behavior. Usually that will mean setting labels to a `numberOfLines` that is 0.
```
Comment (ianegordon) [inline contributing/checklist.md]: Consider:
setting a label's 'numberOfLines' to '0'.

File: contributing/checklist.md [hunk 3]
```diff
@@ -174,6 +174,21 @@ See Apple's [Accessibility Programming Guide for iOS](https://developer.apple.co
 1. Test your control on a device in VoiceOver mode and ensure the bahavior is at least as robust as UIKit.
 1. Enter YES, NO or N/A
 
+### Dynamic Type Support (If possible)
+
+
+Any component that has text, interacts with text, or lays itself out in relation to text must implement Dynamic Type.
+See Apple's [Building Apps with Dynamic Type video](https://developer.apple.com/videos/play/wwdc2017/245/) for further information.
+
+1. Set all fonts with a text style enum like `MDCFontTextStyle` or `UIFontTextStyle`, or via `UIFontMetrics`.
```
Comment (ianegordon) [inline contributing/checklist.md]: Consider:
Set fonts with [UIFont preferredTextForStyle:]  or [UIFont mdc_perferredTextForMaterialStyle:].

**Review by ianegordon — APPROVED** [review]
Please consider comments before merge.

File: contributing/checklist.md (hunk 1 above)
Comment (willlarche) [reply contributing/checklist.md]: Done

File: contributing/checklist.md (hunk 3 above)
Comment (willlarche) [reply contributing/checklist.md]: Done. I kept the font metrics thing.

File: contributing/checklist.md (hunk 2 above)
Comment (willlarche) [reply contributing/checklist.md]: Done.

**Comment by willlarche** [comment]
Travis cancelled because it's documentation file only.

**Pull request merged**